### PR TITLE
Feature branch

### PR DESCRIPTION
**Pull request by geraldgsh**
- finalized project

### CODE REVIEW

File: app/helpers/sessions_helper.rb [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+  def signed_in?
+    !current_user.nil?
+  end
+
+  def sign_out
+    current_user.update_attribute(:remember_token,
+                                  User.digest(User.new_remember_token))
+    cookies.delete(:remember_token)
+    self.current_user = nil
+  end
```
Comment (mcrd25) [inline app/helpers/sessions_helper.rb]: You seem to have application logic inside of your helpers. Helper methods ought to be view specific; the fact that you have to import the helpers in Application Controller for you to have access to those methods in controllers while you do not even use it in your views reiterates this. The views on the other hand, automatically have access to helper methods. 
So, what you can do is add all methods which are not views specific to your application controller directly. What you can also do if your method list gets long, you may use concerns or a lib folder to separate methods by function, etc.

Basically helpers = views, controller = controller. Why write methods in helpers only to have to include it in controller after. Helpers are automatically accessible in views. there is a reason helpers are outside the context of the controller folder. You can, however, use concerns in your controllers.

Read [this article](http://codefol.io/posts/Where-Do-I-Put-My-Code/) for a clearer explanation. 🙂

File: app/helpers/sessions_helper.rb [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+  def current_user=(user)
+    @current_user = user
+  end
+
+  def current_user
+    remember_token = User.digest(cookies[:remember_token])
+    @current_user ||= User.find_by(remember_token: remember_token)
+  end
+
+  def signed_in?
```
Comment (mcrd25) [inline app/helpers/sessions_helper.rb]: This is the only method that really belongs here as it is views specific

File: app/helpers/sessions_helper.rb (hunk 1 above)
Comment (mcrd25) [reply app/helpers/sessions_helper.rb]: You can see this requirement in following screenshot of project's [pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/49722686/) link

Screen Shot 2019-11-20 at 10 48 19 PM

**Comment by geraldgsh** [comment]
Thank you @mcrd25 for taking time to explain about the correct the use of helpers. We now understand that it is solely meant to be used in conjunction with views. We'll make the amendments and submit it once more fore review.

😃😃 Have a pleasant day ahead of you

**Comment by geraldgsh** [comment]
Thanks @mcrd25  @mcrd25 

I'll proceed to merge

🎊🎉🥳

**Pull request merged**